core/src/main/scala/pureconfig/error/ConfigReaderFailures.scala

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package pureconfig.error
+
+case class ConfigReaderFailures(head: ConfigReaderFailure, tail: Seq[ConfigReaderFailure]) {
```

Comment:
Suggestions:
- Explicitly state that this is intended to be a non-empty collection of failures.
- Make the `tail` a `List` or `Vector` so that it doesn't have surprising performance characteristics. (My preference is `List`)
- Replace `toSeq` with either `toList` or `toVector` based on the direction you go above.